core/src/main/java/net/sourceforge/jnlp/util/PropertiesFile.java

```diff
@@ -216,16 +215,19 @@ public boolean load() {
      * Saves the properties to the file.
      */
     public void store() {
+        if (lockableFile.isReadOnly()) {
+            LOG.warn("Cannot save properties as the file is read only");
+            return;
+        }
         final File file = lockableFile.getFile();
         try (final FileOutputStream s = new FileOutputStream(file)) {
-            file.getParentFile().mkdirs();
             delegate.store(s, header);
             // fsync()
             s.getChannel().force(true);
             lastStore = file.lastModified();
             dirty = false;
         } catch (final IOException ex) {
-            LOG.error(IcedTeaWebConstants.DEFAULT_ERROR_MESSAGE, ex);
+            LOG.error("Failed to store", ex);
```

Comment:
In general I ask myself if throwing exceptions would not make more sense. But this would maybe change the runtime behavior...